CMakeLists.txt

```diff
@@ -81,6 +81,7 @@ endif(CUDA_STATIC_RUNTIME)
 
 target_link_libraries(rmm INTERFACE rmm::Thrust)
 target_link_libraries(rmm INTERFACE spdlog::spdlog_header_only)
+target_link_libraries(rmm INTERFACE dl)
```

Comment:
Is this needed? Quick google shows that dladdr now lives in libc rather than libdl?

Reply:
With stack traces enabled, this was needed to compile the tests (original [comment](https://github.com/rapidsai/rmm/pull/626#issuecomment-816341129)). Keith and I briefly discussed this here: https://github.com/rapidsai/rmm/pull/626#discussion_r632139250.

Can you send me the link where you saw that `dladdr` has moved? All I am seeing from this [link](https://linux.die.net/man/3/dladdr) is:

> Link with -ldl.

Reply:
Nevermind, the docs I found were not for linux -- Solaris and something called illumos.  As I said, it was a quick google.

Reply:
Have you tested that all is well in libcudf, cuML, etc. when this library is linked here?  Note that the other `target_link_libraries` for RMM are all header-only, which is why this one has me worried (RMM is a header-only library).

Reply:
@mdemoret reports cuML builds and tests fine against this PR.